Mathlib/Analysis/MellinInversion.lean

```diff
@@ -0,0 +1,116 @@
+    mellin f s = 𝓕 (fun (u : ℝ) ↦ (Real.exp (-s.re * u) • f (Real.exp (-u)))) (s.im / (2 * π)) :=
+  calc
+    _ = ∫ (u : ℝ), Complex.exp (-s * u) • f (Real.exp (-u)) := by
+      rw [mellin, ← rexp_neg_image_aux, integral_image_eq_integral_abs_deriv_smul
+        MeasurableSet.univ rexp_neg_deriv_aux rexp_neg_injOn_aux]
+      simp [rexp_cexp_aux]
+    _ = ∫ (u : ℝ), Complex.exp (↑(-2 * π * (u * (s.im / (2 * π)))) * I) •
+        (Real.exp (-s.re * u) • f (Real.exp (-u))) := by
+      congr
+      ext u
+      trans Complex.exp (-s.im * u * I) • (Real.exp (-s.re * u) • f (Real.exp (-u)))
+      · conv => lhs; rw [← re_add_im s]
+        rw [neg_add, add_mul, Complex.exp_add, mul_comm, ← smul_eq_mul, smul_assoc]
+        norm_cast
+        push_cast
+        ring_nf
+      congr
+      rw [mul_comm (-s.im : ℂ) (u : ℂ), mul_comm (-2 * π)]
+      have : 2 * (π : ℂ) ≠ 0 := by norm_num; exact pi_ne_zero
+      field_simp
+    _ = _ := by
+      simp [fourierIntegral_eq']
+
+theorem mellin_inv_eq_fourierIntegral_inv (σ : ℝ) (f : ℂ → E) {x : ℝ} (hx : 0 < x) :
+    mellin_inv σ f x =
+    (x : ℂ) ^ (-σ : ℂ) • 𝓕⁻ (fun (y : ℝ) ↦ f (σ + 2 * π * y * I)) (-Real.log x) := calc
+  _ = (x : ℂ) ^ (-σ : ℂ) •
+      (∫ (y : ℝ), Complex.exp (2 * π * (y * (-Real.log x)) * I) • f (σ + 2 * π * y * I)) := by
+    rw [mellin_inv, one_div, ← abs_of_pos (show 0 < (2 * π)⁻¹ by norm_num; exact pi_pos)]
+    have hx0 : (x : ℂ) ≠ 0 := ofReal_ne_zero.mpr (ne_of_gt hx)
+    simp_rw [neg_add, cpow_add _ _ hx0, mul_smul, integral_smul]
+    rw [smul_comm, ← Measure.integral_comp_mul_left]
+    congr! 3
+    rw [cpow_def_of_ne_zero hx0, ← Complex.ofReal_log hx.le]
+    push_cast
+    ring_nf
+  _ = _ := by simp [fourierIntegralInv_eq']
+
+variable [CompleteSpace E]
+
+/-- The inverse Mellin transform of the Mellin transform applied to `x > 0` is x. -/
+theorem mellin_inversion (σ : ℝ) (f : ℝ → E) {x : ℝ} (hx : 0 < x) (hf : MellinConvergent f σ)
+    (hFf : VerticalIntegrable (mellin f) σ) (hfx : ContinuousAt f x) :
+    mellin_inv σ (mellin f) x = f x := by
+  let g := fun (u : ℝ) => Real.exp (-σ * u) • f (Real.exp (-u))
+  replace hf : Integrable g := by
+    rw [MellinConvergent, ← rexp_neg_image_aux, integrableOn_image_iff_integrableOn_abs_deriv_smul
+      MeasurableSet.univ rexp_neg_deriv_aux rexp_neg_injOn_aux] at hf
+    replace hf : Integrable fun (x : ℝ) ↦ cexp (-↑σ * ↑x) • f (rexp (-x)) := by
+      simpa [rexp_cexp_aux] using hf
+    norm_cast at hf
+  replace hFf : Integrable (𝓕 g) := by
+    have h2π : 2 * π ≠ 0 := by norm_num; exact pi_ne_zero
+    haveI : HasDistribNeg ℝ := inferInstance
+    simpa [VerticalIntegrable, mellin_eq_fourierIntegral, mul_div_cancel _ h2π] using
+      hFf.comp_mul_right' h2π
+  replace hfx : ContinuousAt g (-Real.log x) := by
+    refine ContinuousAt.smul (by fun_prop) (ContinuousAt.comp ?_ (by fun_prop))
+    simpa [Real.exp_log hx] using hfx
+  calc
+    _ = mellin_inv σ (fun s ↦ 𝓕 g (s.im / (2 * π))) x := by
```

Comment:
Can you spell out the first and last terms in calc blocks? (i.e., don't start or end with an underscore). Lean doesn't need it, but it helps readers a lot.